src/main/java/br/com/fiap/techchallenge/domain/model/enums/CategoriaEnum.java

```diff
@@ -24,4 +24,14 @@ public static CategoriaEnum fromName(String name) throws CategoriaException {
         Predicate<CategoriaEnum> byName = v -> v.getNome().equalsIgnoreCase(name);
         return Arrays.stream(values()).filter(byName).findAny().orElseThrow(() -> new CategoriaException(ErrosEnum.CATEGORIA_INVALIDA, "Categoria não encontrada."));
     }
+
+    public static CategoriaEnum fromIdCategoria(Long id) {
+        for (CategoriaEnum categoria : CategoriaEnum.values()) {
```

Comment:
Aqui podemos simplificar pra tirar esse for:
`Predicate<CategoriaEnum> byId = e -> e.getIdCategoria().equals(id);`
`Arrays.stream(CategoriaEnum.values()).filter(byId).findAny().orElseThrown(() -> new IllegalArgumentException("Invalid codigo: " + id));`